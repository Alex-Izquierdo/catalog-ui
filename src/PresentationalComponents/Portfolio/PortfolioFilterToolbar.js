import React from 'react';
import { Toolbar } from '@patternfly/react-core';
import OrderToolbarItem from '../Shared/OrderToolbarItem';
import FilterToolbarItem from '../Shared/FilterToolbarItem';
import '../Shared/toolbarschema.scss';

const PortfolioFilterToolbar = props => (
  <Toolbar style={ { backgroundColor: '#FFFFFF' } }>
    <FilterToolbarItem { ...props } placeholder={ 'Find a Product' }/>
    <OrderToolbarItem { ...props }/>
  </Toolbar>
);

export default PortfolioFilterToolbar;
