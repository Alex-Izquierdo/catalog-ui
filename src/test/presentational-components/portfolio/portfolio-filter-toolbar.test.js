import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PortfolioFilterToolbar from '../../../presentational-components/Portfolio/PortfolioFilterToolbar';

describe('<PortfolioFilterToolbar />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      addProductsRoute: '/add',
      isLoading: false,
      editPortfolioRoute: '/edit',
      removePortfolioRoute: '/remove-portfolio',
      removeProductsRoute: '/remove-products',
      onFilterChange: jest.fn()
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<PortfolioFilterToolbar { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
