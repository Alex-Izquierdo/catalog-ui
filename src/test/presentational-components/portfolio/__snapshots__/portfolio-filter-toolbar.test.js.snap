// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioFilterToolbar /> should render correctly 1`] = `
<Toolbar
  className={null}
>
  <ToolbarGroup
    className={null}
  >
    <ToolbarItem
      className={null}
    >
      <FilterToolbarItem
        onFilterChange={[MockFunction]}
        placeholder="Filter by name..."
        searchValue=""
      />
    </ToolbarItem>
  </ToolbarGroup>
  <ToolbarGroup
    className={null}
  >
    <ToolbarItem
      className={null}
    >
      <Link
        to="/add"
      >
        <Button
          aria-label="Add Products to Portfolio"
          className=""
          component="button"
          isActive={false}
          isBlock={false}
          isDisabled={false}
          isFocus={false}
          isHover={false}
          type="button"
          variant="primary"
        >
          Add products
        </Button>
      </Link>
    </ToolbarItem>
  </ToolbarGroup>
  <ToolbarGroup
    className={null}
  >
    <ToolbarItem
      className={null}
    >
      <Link
        to="/remove-products"
      >
        <Button
          aria-label="Remove Products from Portfolio"
          className="destructive-color"
          component="button"
          isActive={false}
          isBlock={false}
          isDisabled={false}
          isFocus={false}
          isHover={false}
          type="button"
          variant="link"
        >
          Remove products
        </Button>
      </Link>
    </ToolbarItem>
    <ToolbarItem
      className={null}
    >
      <Dropdown
        className=""
        direction="down"
        dropdownItems={
          Array [
            <Item
              aria-label="Share Portfolio"
              className=""
              component="a"
              href="#"
              isDisabled={false}
              isHovered={false}
            >
              <Link
                className="pf-c-dropdown__menu-item"
                role="link"
              >
                Share
              </Link>
            </Item>,
            <Separator
              className=""
              component="a"
              href="#"
              isDisabled={false}
              isHovered={false}
            />,
            <Item
              aria-label="Edit Portfolio"
              className=""
              component="a"
              href="#"
              isDisabled={false}
              isHovered={false}
            >
              <Link
                className="pf-c-dropdown__menu-item"
                role="link"
                to="/edit"
              >
                Edit
              </Link>
            </Item>,
            <Item
              aria-label="Remove Portfolio"
              className=""
              component="a"
              href="#"
              isDisabled={false}
              isHovered={false}
            >
              <Link
                className="pf-c-dropdown__menu-item destructive-color"
                role="link"
                to="/remove-portfolio"
              >
                Delete
              </Link>
            </Item>,
          ]
        }
        isOpen={false}
        isPlain={true}
        onSelect={[Function]}
        onToggle={[Function]}
        position="right"
        toggle={
          <Kebab
            aria-label="Actions"
            className=""
            id=""
            isActive={false}
            isFocused={false}
            isHovered={false}
            isOpen={false}
            isPlain={false}
            onToggle={[Function]}
            parentRef={null}
          />
        }
      />
    </ToolbarItem>
  </ToolbarGroup>
</Toolbar>
`;
